### PR TITLE
Api side changes for updation of district through account_details page

### PR DESCRIPTION
**Pull request by abhinav098**
Hi team, this PR contains the API side changes for the district updation to user's account through account details page.
Please review and comment.
Thanks.

### CODE REVIEW

File: lib/classes/organisations_user_builder.rb [hunk 1]
```diff
@@ -15,6 +15,8 @@ def initialize(params)
     @user = @organisations_user.user if @organisations_user
     @organisation_id = params["organisation_id"].presence.try(:to_i)
     @user_attributes = params['user_attributes']
+    @user_address_attributes = params['user_address_attributes']
+    @district = District.find(@user_address_attributes['district_id'].presence&.to_i) rescue nil
```
Comment (namrataukirde) [inline lib/classes/organisations_user_builder.rb]: We can use find_by_id here and remove rescue nil.

File: lib/classes/organisations_user_builder.rb [hunk 2]
```diff
@@ -48,10 +50,14 @@ def organisation
   end
 
   def update_user
+    update_user_address if @user_address_attributes['district_id'] && @district
     @user.roles << charity_role unless @user.roles.include?(charity_role)
     @user.update(@user_attributes)
   end
 
+  def update_user_address
+    @user.address.update(district_id: @user_address_attributes['district_id'])
```
Comment (namrataukirde) [inline lib/classes/organisations_user_builder.rb]: We need to verify scenario where  user do not have existing address record.

File: lib/classes/organisations_user_builder.rb (hunk 1 above)
Comment (abhinav098) [reply lib/classes/organisations_user_builder.rb]: done already

File: lib/classes/organisations_user_builder.rb (hunk 2 above)
Comment (abhinav098) [reply lib/classes/organisations_user_builder.rb]: Hi @namrataukirde , Address gets created as soon as user is created, but however, i have added an edge case if user doesn't have address, in that case we'll create the address with the same params, coming from Client.

**Pull request closed without merging**